lib/matplotlib/rcsetup.py

```diff
@@ -175,17 +175,17 @@ def validate_string_or_None(s):
         raise ValueError('Could not convert "%s" to string' % s)
 
 
-def _validate_stringlist_or_string(s):
-    """convert s to string or raise"""
+def _validate_tex_preamble(s):
     if s is None or s == 'None':
         return ""
     try:
         if isinstance(s, str):
             return s
-        if isinstance(s, Iterable):
-            return '\n'.join([str(i) for i in s])
-        raise ValueError()
-    except ValueError:
+        elif isinstance(s, Iterable):
+            return '\n'.join(s)
+        else:
+            raise TypeError
```

Comment:
The try-except seem unnecessary. Afaics, this can only come from the explicit raise and thus can be simplified.

Reply:
No, it can also happen if s is not an iterable of strings.

Reply:
Ok.